### PR TITLE
[Params] Refactors params usage in the admin api controllers

### PR DESCRIPTION
**Pull request by jsmadis**
**What this PR does / why we need it**:

Refactors usage of params to have controllers more secure.

### CODE REVIEW

File: app/concerns/metric_params.rb [hunk 1]
```diff
@@ -10,11 +10,11 @@ module MetricParams
     private
 
     def create_params
-      params.fetch(:metric).permit(DEFAULT_PARAMS | %i[system_name])
+      params.fetch(:metric).permit(DEFAULT_PARAMS | %i[system_name]).to_h
```
Comment (hallelujah) [inline app/concerns/metric_params.rb:13]: I do not get why we need `to_h`
Could you explain?

Comment (jsmadis) [reply app/concerns/metric_params.rb]: I have made a mistake with first assumption of the issue. It's not necessary

File: app/controllers/admin/fields_definitions_controller.rb [hunk 2]
```diff
@@ -76,8 +75,14 @@ def sort
 
   private
 
+  DEFAULT_PARAMS = %i[target label required hidden read_only choices_for_views].freeze
+
   def field_definition_params
-    params[:fields_definition] || {}
+    params.require(:fields_definition).permit(DEFAULT_PARAMS | %i[name])
```
Comment (akostadinov) [inline app/controllers/admin/fields_definitions_controller.rb:81]: `field_definition_params` is also used in `#new`. I guess that `:fields_definition` will not be provided in such call so the `require` call will fail. Have you checked that?

btw, will it be safe to remove `attr_protected :account_id, :tenant_id` from `app/models/fields_definition.rb`?

File: app/controllers/admin/api/buyers_applications_controller.rb [hunk 3]
```diff
@@ -282,11 +282,12 @@ def application_plan
   end
 
   def application_params
-    flat_params.slice(*application_attributes)
+    flat_params.permit(application_attributes | %w[user_key application_id plan_id])
```
Comment (akostadinov) [inline app/controllers/admin/api/buyers_applications_controller.rb:285]: Seems like `plan_id` was not needed in the past. Would you explain why add it now?

**Pull request closed without merging**